[Bug]:  webkit2gtk-4.1 build failure

### Problem description

2024-10-12T16:00:40.3933696Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:60: syntax error, unexpected VOID in '        char _dummy_pkey[(__alignof__(void *) < sizeof(short) ? sizeof(short) : __alignof__(void *))];' at 'void'
2024-10-12T16:00:40.3940285Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:62: syntax error, unexpected '}' in '      } _addr_pkey;' at '}'
2024-10-12T16:00:40.3943337Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:68: syntax error, unexpected '}' in '    };' at '}'
2024-10-12T16:00:40.3946031Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:79: syntax error, unexpected '}' in '};' at '}'
2024-10-12T16:00:40.3949353Z ../build/WebKitGTK/DerivedSources/webkit/WebKitWebView.h:549: Warning: WebKit2: Couldn't find 'run_async_javascript_function_in_world_finish' for the corresponding async function: 'run_async_javascript_function_in_world'
2024-10-12T16:00:40.3951738Z <unknown>:: Fatal: WebKit2: warnings configured as fatal
2024-10-12T16:00:40.3952604Z <unknown>:: Fatal: WebKit2: warnings configured as fatal
2024-10-12T16:00:40.3953153Z 
2024-10-12T16:00:40.5462728Z [6450/6464] Building C object Tools/MiniBrowser/gtk/CMakeFiles/MiniBrowser.dir/BrowserCellRendererVariant.c.o
2024-10-12T16:00:41.5517628Z [6451/6464] Generating WebKit2WebExtension-4.1.gir
2024-10-12T16:00:41.5519158Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:55: syntax error, unexpected VOID in '        char _dummy_bnd[(__alignof__(void *) < sizeof(short) ? sizeof(short) : __alignof__(void *))];' at 'void'
2024-10-12T16:00:41.5521023Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:58: syntax error, unexpected '}' in '      } _addr_bnd;' at '}'
2024-10-12T16:00:41.5523158Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:60: syntax error, unexpected VOID in '        char _dummy_pkey[(__alignof__(void *) < sizeof(short) ? sizeof(short) : __alignof__(void *))];' at 'void'
2024-10-12T16:00:41.5525266Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:62: syntax error, unexpected '}' in '      } _addr_pkey;' at '}'
2024-10-12T16:00:41.5526729Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:68: syntax error, unexpected '}' in '    };' at '}'
2024-10-12T16:00:41.5528110Z /home/builder/.termux-build/_cache/android-r27b-api-24-v1/sysroot/usr/include/asm-generic/siginfo.h:79: syntax error, unexpected '}' in '};' at '}'
2024-10-12T16:00:42.1758973Z [6452/6464] Building CXX object Source/WebCore/CMakeFiles/WebCoreTestSupport.dir/__/__/WebCore/DerivedSources/JSInternalSettingsGenerated.cpp.o
2024-10-12T16:00:42.1760411Z ninja: build stopped: subcommand failed.
2024-10-12T16:00:42.2442589Z ##[error]Process completed with exit code 1.



[job-logs.txt](https://github.com/user-attachments/files/17350697/job-logs.txt)


### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

```shell
no need
```

--- a/x11-packages/webkit2gtk-4.1/build.sh
+++ b/x11-packages/webkit2gtk-4.1/build.sh
@@ -3,7 +3,7 @@ TERMUX_PKG_DESCRIPTION="A full-featured port of the WebKit rendering engine"
 TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="2.42.4"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://webkitgtk.org/releases/webkitgtk-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=52288b30bda22373442cecb86f9c9a569ad8d4769a1f97b352290ed92a67ed86
 TERMUX_PKG_DEPENDS="atk, enchant, fontconfig, freetype, glib, gst-plugins-bad, gst-plugins-base, gst-plugins-good, gstreamer, gtk3, harfbuzz, harfbuzz-icu, libc++, libcairo, libgcrypt, libhyphen, libicu, libjpeg-turbo, libpng, libsoup3, libtasn1, libwebp, libxml2, libx11, libxcomposite, libxdamage, libxslt, libxt, littlecms, openjpeg, pango, woff2, zlib"
@@ -42,13 +42,8 @@ termux_step_post_get_source() {
 termux_step_pre_configure() {
 	termux_setup_gir
 
-	# FIXME: `GI_VERSION` mismatched from Termux and the building machine. On April 22,
-	# FIXME: 2024, Termux has version 1.80.1 but Ubuntu 22.04 has version 1.72.0.
-	# FIXME: `cmake` will pick up `GI_VERSION` from the config files of Termux, but
-	# FIXME: it is intended to use the version in the building machine.
-	if [ "$TERMUX_ON_DEVICE_BUILD" = false ]; then
-		sed -i 's@if ("${GI_VERSION}" VERSION_GREATER_EQUAL 1.79.2)@if (FALSE)@g' \
-			$TERMUX_PKG_SRCDIR/Source/WebKit/PlatformGTK.cmake
+	if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
+		export CXXFLAGS+=" -Wno-missing-template-arg-list-after-template-kw"
 	fi
 
 	# Workaround for https://github.com/android/ndk/issues/1973

--- a/x11-packages/webkitgtk-6.0/build.sh
+++ b/x11-packages/webkitgtk-6.0/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="A full-featured port of the WebKit rendering engine"
 TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="2.42.4"
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://webkitgtk.org/releases/webkitgtk-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=52288b30bda22373442cecb86f9c9a569ad8d4769a1f97b352290ed92a67ed86
 TERMUX_PKG_DEPENDS="enchant, fontconfig, freetype, glib, gst-plugins-bad, gst-plugins-base, gst-plugins-good, gstreamer, gtk4, harfbuzz, harfbuzz-icu, libc++, libcairo, libgcrypt, libhyphen, libicu, libjpeg-turbo, libpng, libsoup3, libtasn1, libwebp, libxml2, libx11, libxcomposite, libxdamage, libxslt, libxt, littlecms, openjpeg, pango, woff2, zlib"
@@ -35,16 +36,11 @@ termux_step_pre_configure() {
 		-DENABLE_WEBDRIVER=OFF
 	"
 
-	# FIXME: `GI_VERSION` mismatched from Termux and the building machine. On April 22,
-	# FIXME: 2024, Termux has version 1.80.1 but Ubuntu 22.04 has version 1.72.0.
-	# FIXME: `cmake` will pick up `GI_VERSION` from the config files of Termux, but
-	# FIXME: it is intended to use the version in the building machine.
-	if [ "$TERMUX_ON_DEVICE_BUILD" = false ]; then
-		sed -i 's@if ("${GI_VERSION}" VERSION_GREATER_EQUAL 1.79.2)@if (FALSE)@g' \
-			$TERMUX_PKG_SRCDIR/Source/WebKit/PlatformGTK.cmake
-	fi
-
 	termux_setup_gir
+
+	if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
+		export CXXFLAGS+=" -Wno-missing-template-arg-list-after-template-kw"
+	fi
 
 	# Workaround for https://github.com/android/ndk/issues/1973
 	[ "$TERMUX_ARCH" == "arm" ] && sed -i '/#define MUST_TAIL_CALL \[\[clang::musttail]]/d' Source/WTF/wtf/Compiler.h

